ElementX/Sources/Screens/ManageRoomMemberSheet/View/ManageRoomMemberSheetView.swift

```diff
@@ -28,22 +37,34 @@ struct ManageRoomMemberSheetView: View {
             }
             
             Section {
-                if context.viewState.canKick {
+                if context.viewState.permissions.canKick {
                     ListRow(label: .default(title: L10n.screenBottomSheetManageRoomMemberRemove,
                                             icon: \.close,
                                             role: .destructive),
                             kind: .button {
                                 context.send(viewAction: .kick)
                             })
+                            .disabled(context.viewState.isKickDisabled)
                 }
                 
-                if context.viewState.canBan {
-                    ListRow(label: .default(title: L10n.screenBottomSheetManageRoomMemberBan,
-                                            icon: \.block,
-                                            role: .destructive),
-                            kind: .button {
-                                context.send(viewAction: .ban)
-                            })
+                if context.viewState.permissions.canBan {
+                    if context.viewState.isMemberBanned {
+                        ListRow(label: .default(title: L10n.screenBottomSheetManageRoomMemberUnban,
+                                                icon: \.restart,
+                                                role: .destructive),
+                                kind: .button {
+                                    context.send(viewAction: .unban)
+                                })
+                                .disabled(context.viewState.isBanUnbanDisabled)
+                    } else {
+                        ListRow(label: .default(title: L10n.screenBottomSheetManageRoomMemberBan,
+                                                icon: \.block,
+                                                role: .destructive),
+                                kind: .button {
+                                    context.send(viewAction: .ban)
+                                })
+                                .disabled(context.viewState.isBanUnbanDisabled)
+                    }
```

Comment:
Pleeease can we use ternaries rather than duplicating the view layout to its easier to read the view 🥺

(Ideally putting the title/icon etc in the view state as banUnbanTitle etc)
```suggestion
                    let title = context.viewState.isMemberBanned ? L10n.screenBottomSheetManageRoomMemberUnban : L10n.screenBottomSheetManageRoomMemberBan
                    let icon: KeyPath = context.viewState.isMemberBanned ? \.restart : \.block
                    let action: ViewAction = context.viewState.isMemberBanned ? .unban : .ban
                    
                    ListRow(label: .default(title: title,
                                            icon: icon,
                                            role: .destructive),
                            kind: .button {
                                context.send(viewAction: action)
                            })
                                .disabled(context.viewState.isBanUnbanDisabled)
```